### DISCUSSION TITLE
Negative mass/concentration problem

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#23099] [Q&A Modules: General] [answered]
Dear all, 

I am stuck with a numerical problem with our [application](https://github.com/Domino2022/tiger.git). Here I want to show one simple model to ask a question. 

I have a 1D model and let water flow through the 1D system by setting a constant water flow rate at one boundary. In addition, I want to have a chemical component following the water to flow through the 1D system by giving a constant concentration at the boundary with water injection. Initially, there is only water in the system. 

Now the question is after computation, there is always a negative value for chemical component concentration (please see figure below), even at the first time step. 

I tried to check my code and solver conditions but nothing helps me with the negative value. Could you help to give some comments regarding that? Thanks in advance. Attached are my [input file and results](https://github.com/Joseph-0123/Negative-mass-concentration-problem.git).

![withBC_on_Right_Step2 (0 2s)](https://user-images.githubusercontent.com/56687392/211151532-ce28e40c-a0f0-4725-b0fd-0750359cfdd9.JPG)

**Comment by GiudGiud** [comment, marked as answer]
Hello

You have boundary conditions on both ends for this solute right?
It doesnt seem physically to have an inlet at a concentration and an outlet at 0.
Is the solute supposed to have entirely reacted or something? I dont see a loss/reaction kernel

Guillaume

**Reply by Joseph-0123** [reply]
> Hello
> 
> You have boundary conditions on both ends for this solute right? It doesnt seem physically to have an inlet at a concentration and an outlet at 0. Is the solute supposed to have entirely reacted or something? I dont see a loss/reaction kernel
> 
> Guillaume

Yes. I have boundary conditions on both ends for this solute. 
If I only set the inlet at a concentration and do not set an outlet, I still got similar results with some negative concentrations (shown in the attached picture). 

Only the advection of concentration is considered in this process. There are no loss/reaction kernels.
![noBC_on_Right_Step2 (0 2s)](https://user-images.githubusercontent.com/56687392/211151761-b7a8e1c7-6448-4ce3-8ec8-85af5fa96a4b.JPG)

**Reply by GiudGiud** [reply]
What is the advection velocity here?
The solute concentration is essentially 0 here. So the advection and the diffusion are not propagating the solute

**Reply by GiudGiud** [reply]
while some numbers are negative, they are mostly 0

**Reply by Joseph-0123** [reply]
> What is the advection velocity here? The solute concentration is essentially 0 here. So the advection and the diffusion are not propagating the solute

The advection velocity value is 3.5e-6 m/s, which is the artificial darcy velocity. In the first time step (t= 0.1 s), some solute values are negative and close to 0 but the minimum value is -0.15. Also, as time goes on （i.e., 363.4s）, the minimum solute concentration can be minus several hundred (-545. 818, shown in the attached pictures).

![noBC_on_Right_Step3634 (363 4s)](https://user-images.githubusercontent.com/56687392/211155205-bbd6e262-60cb-426b-820c-a8be57d9368a.JPG)

**Reply by GiudGiud** [reply]
ok this latest screenshot does show this more clearly. 

Is this with the two boundary conditions on both ends? As I was mentioning earlier, this seems unphysical.
Or with the single boundary condition on one end?
